### DISCUSSION TITLE
Capture first node

### DISCUSSION THREAD

**Discussion by dchinmay2** [nvim-treesitter/nvim-treesitter#3703] [Q&A] [answered]
I'm trying to detect hashbangs, how do I capture the first node of the tree in a query?

**Comment by ObserverOfTime** [comment, marked as answer]
Here's an example from Perl:
https://github.com/nvim-treesitter/nvim-treesitter/blob/b9bcbf8d73b5a6c3e04922936b5fc500b436d4f5/queries/perl/highlights.scm#L88-L89

**Reply by dchinmay2** [reply]
Thanks, this was it (fish doesn't have `source_file` but it was easy to find the correct one i.e. `program` by looking at the parser repo)
